chain/consensus/mir/types.go

```diff
@@ -0,0 +1,34 @@
+package mir
+
+import (
+	logging "github.com/ipfs/go-log/v2"
+)
+
+const (
+	// ConfigOffset is the number of epochs by which to delay configuration changes.
+	// If a configuration is agreed upon in epoch e, it will take effect in epoch e + 1 + configOffset.
+	ConfigOffset        = 2
+	TransportType       = 0
+	ReconfigurationType = 1
+)
+
+var log = logging.Logger("mir-consensus")
+
+const (
+	ConfigMessageType = 0 // Mir specific config message
+	SignedMessageType = 1 // Lotus signed message
+)
+
+func NewSignedMessageBytes(msg, opaque []byte) []byte {
+	var payload []byte
+	payload = append(msg, opaque...)
```

Comment:
Opaque was used in Tendermint. We don't need it now

Reply:
Fixed. Removed the opaque and created `NewTypedMessageBytes` that accepts a `MirMsgType` as input.

Reply:
Sorry for nagging, and feel free to ignore and re-resolve.

Wouldn't it be more meaningful to just have one function, something like
```go
func MessageBytes(msg <whatever_the_type_of_the_message_is>) []byte {
	data := msg.Serialize()
	return append(msg, msg.Type)
}
```
I assume here that the `Serialize` method of the message itself is hard to change / not written by us. Of course, ideally this logic would be encapsulated there...

Reply:
Fixed. Added a `MirMessage` interface with a `Serialize` function so that all message types that we want to use with Mir need to implement that interface. That being said, it seems we only use `SignedMessages` for now, we may be able to completely remove this.